### ISSUE TITLE
[Bug]: Backup of german exchange contacts doesn’t work

### ISSUE THREAD

**Issue by lts-developer** [alcionai/corso#4175]
### What happened?

A new backup of the exchange contacts on an empty storage doesn’t work, when the language setting on the website https://outlook.office.com/mail/ (> Settings > General > Language and time > Language) is any other than “English” (tested and doesn’t work with: “German” and “Italian”).
The backup job status says “Completed”, but when you try to restore the contacts with Corso CLI you get “no items match the provided selectors”.
It maybe has something to do with the Outlook-folder-names. In English “Inbox”, “Contacts”, etc. and in German “Posteingang”, „Kontakte“, etc.?
Tested with a new full backup and with incremental backups.

Another language-related problem is: 
When you change the language and “Rename default folders so their names match the specified language” on the website https://outlook.office.com/mail/ the incremental (and maybe even the new full backup, if there is an existing backup of the mailbox on the storage) backup completely fail with the error:
           running backup: merging details: getting updated info for entry: getting new paths: empty location key
(As soon as you switch back to the previously set language the backup is working again as usual.)
Tested with “German” and “Italian”.

./corso backup create exchange --mailbox [address@office365.com](mailto:address@office365.com)
Connecting to repository:      1s done
Connecting to M365:         done
Connecting to M365:         done
Backing Up ∙ [address@office365.com](mailto:address@office365.com)
Discovering items to backup:      4s done
∙ contacts:      0s done
∙ email:      3s done
∙ events:      0s done
Backing up data:      2s done
Error:
        running backup: merging details: getting updated info for entry: getting new paths: empty location key
Completed Backups:
No backups available
Error: 1 of 1 backups failed

### Corso Version?

Corso v0.12.0

### Where are you running Corso?

Linux x86_64 - Debian 11

**Comment by ryanfkeepers** [comment]
Thanks for the report, @lts-developer!  We're digging into the root cause now.

**Comment by ryanfkeepers** [comment]
Alrighty!  Quick follow-up.  We are able to repro both issues.  For the first one (new backup) we also have a root cause: one line of our code is comparing the conatacts folder display name with the graph api's english [well-known folder name](https://learn.microsoft.com/en-us/graph/api/resources/mailfolder?view=graph-rest-1.0) const.  For the second issue (incremental) we don't have a root cause yet, but with a functional repro we should have that figured out in short order.

Once we have fixes in place, we can build the latest version for you to test with and verify the fixes.  If you want the fastest possible response to the issue and its fixes, feel free to join us on discord: https://discord.gg/pqW93kxssf.

**Comment by ryanfkeepers** [comment]
The initial fixes are now in place.  If you could help us out, would you please download and run the [latest build](https://github.com/alcionai/corso/actions/runs/6054940760) to confirm whether we've resolved your issues?

A note on the reproduction: we weren't able to repro the incremental problem on v0.12 alone.  Instead, we had to begin with a v0.12 backup, and then downgrade to v0.11.1 for the incremental backup.  The downgrade itself isn't necessary, per se.  But v0.12 does contain a change that protects against a part of the bug on display.  For our own verification, can you please double check which versions you were running when you encountered the bugs?  Thanks!

**Comment by ntolia** [comment]
@lts-developer Just checking in here to see if we should close this issue out. Did you get a chance to repro with the latest build?

**Comment by lts-developer** [comment]
Thanks for the answer @ryanfkeepers! 
I had it scheduled for today to leave a comment.

In the last few days we have tried several times to change the language on the website https://outlook.office.com/mail and kept doing full + incremental backups. We weren’t able to reproduce the issue “new backup” (contacts) with the latest build (https://github.com/alcionai/corso/actions/runs/6054940760).  

Then we double checked the issue “incremental” with the latest build + with v0.12.0 and we weren’t able to reproduce the issue. Even if you start the full backups with v0.11.1 and upgrade then to v0.12.0 or later, the backups were performed successfully without errors. We got the same results here as you in your tests: the incremental backup fails only with v0.11.1. 

Thank you.
